src/Components/WeatherDetails/WeatherShowMoreLessButton.js

```diff
@@ -0,0 +1,30 @@
+import React, { Component } from 'react';
+
+import { AnimateDetailsButton } from '../../common/Animations/Animations';
+
+class Button extends Component {
+  state = {
+    active: false,
+  };
+
+  handleActiveState = () => {
```

Comment:
handleButtonState

const {active} = this.state;
const {onClick} = this.props;

this.setState({ active: !active}, onClick);

this.setState takes the second argument a callback. If you want to be sure, that your function is ran after state is updated, use callback to setState as above. 
remember, this.setState is asynchronous not synchronous